dpgen/data/gen.py

```diff
@@ -237,12 +237,24 @@
 def poscar_scale(poscar_in, poscar_out, scale):
     with open(poscar_in) as fin:
         lines = list(fin)
-    if "D" == lines[7][0] or "d" == lines[7][0]:
+
+    # Determine if "Selective dynamics" is present
+    if lines[7].strip().lower().startswith("s"):
```

Comment:
Consider using a more pythonic way to check if the line starts with 's' or 'S' by using the `str.startswith()` method. This method can take a tuple of prefixes to check. This would make the code more readable and concise. [important]